bcda/api/requests_test.go

```diff
@@ -123,159 +127,80 @@ func (s *RequestsTestSuite) TestRunoutDisabled() {
+			for _, groupID := range groupIDs {
+				rp := middleware.RequestParameters{
+					Version:       "v1",
+					ResourceTypes: []string{resource},
+					Since:         since,
+				}
+				rr := httptest.NewRecorder()
+				req := s.genGroupRequest(groupID, rp)
+				h.BulkGroupRequest(rr, req)
+				assert.Equal(s.T(), http.StatusAccepted, rr.Code)
+			}
 		}
-
-		u.RawQuery = q.Encode()
-		req := httptest.NewRequest("GET", u.String(), nil)
-		rctx := chi.NewRouteContext()
-		rctx.URLParams.Add("groupId", "all")
-		req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
-
-		s.T().Run(fmt.Sprintf("%s-group", tt.name), func(t *testing.T) {
-			rr := httptest.NewRecorder()
-			h.BulkGroupRequest(rr, req)
-			assert.Equal(t, http.StatusBadRequest, rr.Code)
-			assert.Contains(t, rr.Body.String(), tt.errMsg)
-		})
-
-		s.T().Run(fmt.Sprintf("%s-patient", tt.name), func(t *testing.T) {
-			rr := httptest.NewRecorder()
-			h.BulkGroupRequest(rr, req)
-			assert.Equal(t, http.StatusBadRequest, rr.Code)
-			assert.Contains(t, rr.Body.String(), tt.errMsg)
-		})
 	}
-}
 
-func (s *RequestsTestSuite) TestCheck429() {
-	validJob := models.Job{RequestURL: "/api/v1/Group/$export", Status: models.JobStatusInProgress, CreatedAt: time.Now()}
-	expiredJob := models.Job{RequestURL: "/api/v1/Group/$export", Status: models.JobStatusInProgress, CreatedAt: time.Now().Add(-2 * GetJobTimeout())}
-	duplicateType := models.Job{RequestURL: "/api/v1/Group/$export?_type=Patient", Status: models.JobStatusInProgress, CreatedAt: time.Now()}
-	tests := []struct {
-		name        string
-		job         models.Job
-		version     string
-		passesCheck bool
-	}{
-		{"Same version", validJob, "v1", false},
-		{"Different version", validJob, "v2", true},
-		{"Invalid job (bad URL)", models.Job{RequestURL: string([]byte{0x7f})}, "", false},
-		{"Invalid job (no version)", models.Job{RequestURL: "/api/Group/$export"}, "", false},
-		{"Same version - expired", expiredJob, "v1", true},
-		{"Duplicate type", duplicateType, "v1", false},
-	}
-
-	for _, tt := range tests {
-		s.T().Run(tt.name, func(t *testing.T) {
-			res, err := check429([]*models.Job{&tt.job}, []string{"Patient"}, tt.version)
-			if tt.passesCheck {
-				assert.NotNil(t, res)
-				assert.NoError(t, err)
-			} else {
-				assert.Nil(t, res)
-				assert.Error(t, err)
+	// Validate patient requests
+	for _, resource := range resources {
+		for _, since := range sinces {
+			rp := middleware.RequestParameters{
+				Version:       "v1",
+				ResourceTypes: []string{resource},
+				Since:         since,
 			}
-		})
+			rr := httptest.NewRecorder()
+			h.BulkPatientRequest(rr, s.genPatientRequest(rp))
+			assert.Equal(s.T(), http.StatusAccepted, rr.Code)
+		}
 	}
 }
 
-// TestBulkRequestWithOldJobPaths verifies that bulk requests for a caller with older jobs
-// whose paths are no longer valid will still be accepted.
-func (s *RequestsTestSuite) TestBulkRequestWithOldJobPaths() {
```

Comment:
Replace by test in [ratelimit_test.go](https://github.com/CMSgov/bcda-app/pull/673/files#diff-75b0d1f2ed7de3581ea31fc1ebe5a2747d66a4e5f65fe8034494cc040b131d39R46)